File: src/dotnet/Datack.Common/Models/Data/Job.cs
using System.ComponentModel.DataAnnotations;
using Datack.Common.Models.Internal;

namespace Datack.Common.Models.Data;

public class Job
{
    [Key]
    public Guid JobId { get; set; }

    public String Name { get; set; }

    public Boolean IsActive { get; set; }

    public String Group { get; set; }

    public Int32 Priority { get; set; }

    public String Description { get; set; }

    public String Cron { get;set; }

    public Int32?  DeleteLogsTimeSpanAmount { get; set; }

    public String DeleteLogsTimeSpanType { get; set; }
        
    public JobSettings Settings { get; set; }
}

File: src/react/src/models/job.ts
export type Job = {
  jobId: string;
  name: string;
  isActive: boolean;
  group: string;
  priority: number;
  cron: string;
  description: string;
  deleteLogsTimeSpanAmount: number | null;
  deleteLogsTimeSpanType: string | null;
  settings: JobSettings;
};

export type JobSettings = {
  emailOnError: boolean;
  emailOnSuccess: boolean;
  emailTo: string | null;
};
